src/main/java/com/avaje/ebeaninternal/server/persist/dml/InsertMeta.java

```diff
@@ -18,6 +19,9 @@
  */
 public final class InsertMeta {
 
+  private enum IdGeneration {
+    BY_SERVER, BY_CLIENT, BY_CLIENT_WITH_IDENTIY_INSERT;
```

Comment:
I'm unconvinced by this part/change.  We should just adopt sequences (with a good increment by) and get full JDBC batch support.  If SqlServer ever supports Identity with getGeneratedKeys using JDBC batch then we revisit (but it hasn't happened and at this point (some years later) I suspect it won't ... so instead adopt sequences and operate like Oracle for JDBC batching inserts).